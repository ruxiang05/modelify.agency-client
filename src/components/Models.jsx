import React from 'react';
import PropTypes from 'prop-types';
import ModelsList from './ModelList';
import PageHeader from './PageHeader';
import addModelIcon from '../assets/icons/user-plus.svg';

class Models extends React.Component {
  constructor(props) {
    super(props);
    this.redirectToAddModel = this.redirectToAddModel.bind(this);
  }

  redirectToAddModel() {
    const { history } = this.props;
    history.push('/add-model');
  }

  render() {
    return (
      <React.Fragment>
        <PageHeader
          title="Models"
          actionIcon={addModelIcon}
          action={this.redirectToAddModel}
        />
        <div className="page">
          <ModelsList />
        </div>
      </React.Fragment>
    );
  }
}

Models.propTypes = {
  history: PropTypes.shape({}).isRequired,
};

export default Models;
