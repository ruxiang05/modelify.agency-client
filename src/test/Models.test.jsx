import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import Models from '../components/Models';

const history = createBrowserHistory({
  basename: '/',
  keyLength: 0,
});
let wrapper;

beforeEach(() => {
  wrapper = shallow(<Models history={history} />);
});

describe('Models', () => {
  it('should render ModelList', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect to /add-model', () => {
    wrapper.instance().redirectToAddModel();
    expect(history.location.pathname).toBe('/add-model');
  });
});
