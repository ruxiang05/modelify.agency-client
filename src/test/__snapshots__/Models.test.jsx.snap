// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Models should render ModelList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Models
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader
          action={[Function]}
          actionIcon="user-plus.svg"
          backButton={false}
          title="Models"
        />,
        <div
          className="page"
        >
          <ModelList />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": [Function],
          "actionAltText": undefined,
          "actionIcon": "user-plus.svg",
          "backButton": false,
          "goBackAction": undefined,
          "history": undefined,
          "title": "Models",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ModelList />,
          "className": "page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            action={[Function]}
            actionIcon="user-plus.svg"
            backButton={false}
            title="Models"
          />,
          <div
            className="page"
          >
            <ModelList />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": [Function],
            "actionAltText": undefined,
            "actionIcon": "user-plus.svg",
            "backButton": false,
            "goBackAction": undefined,
            "history": undefined,
            "title": "Models",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ModelList />,
            "className": "page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
